Source/engine/render/text_render.cpp

```diff
@@ -288,13 +288,17 @@ std::string WordWrapString(string_view text, size_t width, GameFontTables size,
 			continue;
 		}
 
-		uint8_t frame = codepoint & 0xFF;
-		uint32_t unicodeRow = codepoint >> 8;
-		if (unicodeRow != currentUnicodeRow || kerning == nullptr) {
-			kerning = LoadFontKerning(size, unicodeRow);
-			currentUnicodeRow = unicodeRow;
+		if (codepoint == ZWSP) {
+			lineWidth += spacing;
+		} else {

```

Comment:
ZWSP should not add spacing.

```suggestion
		if (codepoint != ZWSP) {
```

Reply:
Hmm, it doesn't look good with this change.

Reply:
Odd, I guess spacing is needed somehow:

![image](https://user-images.githubusercontent.com/216339/140676786-07fe6e74-4637-4b55-b7ee-5f1bd34da566.png)

Reply:
Anyway, this is clearly the correct change, so I've applied it.
The text going all the way to the edge bug must be elsewhere.

Reply:
I might actually have forgotten to adjust the CJK width after we did the final render of the font, I'll have a look sometime this week.

Reply:
OK, sounds like it's not a blocker for merging this! :+1:

Reply:
This was fixed by #3457